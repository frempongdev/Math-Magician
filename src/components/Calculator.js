import React, { useState } from 'react';
import calculate from '../logic/calculate';

const initialState = { total: 0, next: '', operation: '' };
const Calculator = () => {
  const [state, setState] = useState(initialState);

  const handleNumberClick = (e) => {
    const response = calculate(state, e.target.textContent);
    setState(response);
  };

  return (
    <div className="calcMain">
      <h2 className="letsDo">Let&#39;s do some math&#33;</h2>
      <div className="calc-box">
        <div className="calc-div">
          <div id="answer" className="answer">
            {state.total}
            {state.operation}
            {state.next}
          </div>
          <button type="button" onClick={handleNumberClick} id="reset" className="special">AC</button>
          <button type="button" onClick={handleNumberClick} id="plus-or-minus" className="special">+/-</button>
          <button type="button" onClick={handleNumberClick} id="percent" className="special">%</button>
          <button type="button" onClick={handleNumberClick} id="divide-by" className="operand">รท</button>
          <button type="button" onClick={handleNumberClick} id="7" className="num">7</button>
          <button type="button" onClick={handleNumberClick} id="8" className="num">8</button>
          <button type="button" onClick={handleNumberClick} id="9" className="num">9</button>
          <button type="button" onClick={handleNumberClick} id="times" className="operand">x</button>
          <button type="button" onClick={handleNumberClick} id="4" className="num">4</button>
          <button type="button" onClick={handleNumberClick} id="5" className="num">5</button>
          <button type="button" onClick={handleNumberClick} id="6" className="num">6</button>
          <button type="button" onClick={handleNumberClick} id="minus" className="operand">-</button>
          <button type="button" onClick={handleNumberClick} id="1" className="num">1</button>
          <button type="button" onClick={handleNumberClick} id="2" className="num">2</button>
          <button type="button" onClick={handleNumberClick} id="3" className="num">3</button>
          <button type="button" onClick={handleNumberClick} id="plus" className="operand">+</button>
          <button type="button" onClick={handleNumberClick} id="zero" className="num">0</button>
          <button type="button" onClick={handleNumberClick} id="dot" className="special">.</button>
          <button type="button" onClick={handleNumberClick} id="equal-to" className="operand">=</button>
        </div>
      </div>
    </div>
  );
};
export default Calculator;
