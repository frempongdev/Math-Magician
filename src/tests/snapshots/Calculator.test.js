import renderer from 'react-test-renderer';
import Calculator from '../../components/Calculator.js';// eslint-disable-line import/extensions

describe('Calculator', () => {
  test('should match with Calculator snapshot', () => {
    const tree = renderer.create(
      <Calculator />,
    );
    expect(tree).toMatchSnapshot();
  });
});
