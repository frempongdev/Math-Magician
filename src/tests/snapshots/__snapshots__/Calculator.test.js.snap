// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should match with Calculator snapshot 1`] = `
<div
  className="calcMain"
>
  <h2
    className="letsDo"
  >
    Let's do some math!
  </h2>
  <div
    className="calc-box"
  >
    <div
      className="calc-div"
    >
      <div
        className="answer"
        id="answer"
      >
        0
      </div>
      <button
        className="special"
        id="reset"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="special"
        id="plus-or-minus"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="special"
        id="percent"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="operand"
        id="divide-by"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="num"
        id="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="num"
        id="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="num"
        id="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="operand"
        id="times"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="num"
        id="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="num"
        id="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="num"
        id="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="operand"
        id="minus"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="num"
        id="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="num"
        id="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="num"
        id="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="operand"
        id="plus"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="num"
        id="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="special"
        id="dot"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operand"
        id="equal-to"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
